src/linear/coordinate_common.h

```diff
@@ -223,7 +223,15 @@ class FeatureSelector {
   virtual void Setup(const gbm::GBLinearModel &model,
                      const std::vector<GradientPair> &gpair,
                      DMatrix *p_fmat,
-                     float alpha, float lambda, int param) {}
+                     float alpha, float lambda, int param) {
+    // Remove unused parameter compiler warning.
```

Comment:
Can we just remove the parameter in function definition:

```
void Setup(const gbm::GBLinearModel&, std::vector<GradientPair> const&, ...)
```

Reply:
I did not do it for "documentation" purpose, if removing arguments is fine, shall I do it for the full PR?

Reply:
@vcarpani No, @trivialfis means that you should leave out the name of each argument, not the argument itself, e.g.
```cpp
void Setup(const gbm::GBLinearModel&, std::vector<GradientPair> const&, ...)
```

You cannot remove the argument, since this function overrides a virtual function of the parent class.

Reply:
@hcho3, sorry I might have not expressed myself correctly, I understood that I have to remove just the names; I am used to remove the names in `.cpp` files and use `(void)` in `.h` and `.hpp` files, but if removing is fine here, I can do it as well. Shall I do it for all the occasions where I used `(void)`?

Reply:
Yes, removing names is fine in this header.